azurefunction/voice-control/run.csx

```diff
@@ -15,170 +15,189 @@ using NAudio.Wave;
+                    if (int.TryParse(entity, out num))
                     {
-                        return "blink:" + dic[entity];
+                        return "blink:" + num;
                     }
-                    catch (Exception)
+                    else
                     {
-                        //ignore
+                        try
+                        {
+                            return "blink:" + dic[entity];
+                        }
+                        catch (Exception)
+                        {
+                            log.Info("Cannot parse blink times");
+                        }
                     }
                 }
-            }
+                return "None";
 
-            Console.WriteLine("Cannot parse Blink intent");
-            return "None";
-        }
-        if (intent == "Display")
-        {
-            string str = result.query.ToString();
-            str = Regex.Replace(str, "display ", "", RegexOptions.IgnoreCase);
-            Console.WriteLine("Display:" + str);
-            if (str.Length > 50)
-            {
-                str = str.Substring(0, 50);
-            }
-            return "display:" + str;
-        }
-        if (intent == "Sensor")
-        {
-            if (result.entities.Count != 0)
-            {
-                var entity = result.entities[0].entity.ToString();
-                if (entity == "temperature" || entity == "humidity")
-                {
-                    return "sensor:humidtemp";
-                }
-                if (entity == "motion")
-                {
-                    return "sensor:motiongyro";
-                }
-                if (entity == "magnetic")
+            case "Display":
+                var str = result.query.ToString();
+                str = Regex.Replace(str, "display ", "", RegexOptions.IgnoreCase);
+
+                if (str.Length > 50)
                 {
-                    return "sensor:magnetic";
+                    str = str.Substring(0, 50);
                 }
-                if (entity == "pressure")
+
+                return "display:" + str;
+
+            case "Sensor":
+                if (result.entities.Count != 0)
                 {
-                    return "sensor:pressure";
+                    var entity = result.entities[0].entity.ToString();
+                    log.Info("Sensor: " + entity);
+                    if (entity == "temperature" || entity == "humidity")
+                    {
+                        return "sensor:humidtemp";
+                    }
+                    else if (entity == "motion")
+                    {
+                        return "sensor:motiongyro";
+                    }
+                    else if (entity == "magnetic")
+                    {
+                        return "sensor:magnetic";
+                    }
+                    else if (entity == "pressure")
+                    {
+                        return "sensor:pressure";
```

Comment:
maybe you could create a mapping for this, then you could just return "sensor:"+ mapping[entity]